src/plugins/condition/criterion/TelemetryCriterion.js

```diff
@@ -60,20 +67,44 @@ export default class TelemetryCriterion extends EventEmitter {
         };
 
         if (data) {
-            // TODO check back to see if we should format times here
-            this.timeAPI.getAllTimeSystems().forEach(timeSystem => {
+            this.openmct.time.getAllTimeSystems().forEach(timeSystem => {
                 datum[timeSystem.key] = data[timeSystem.key]
             });
         }
         return datum;
     }
 
-    handleSubscription(data) {
-        if(this.isValid()) {
-            this.emitEvent('criterionResultUpdated', this.formatData(data));
-        } else {
-            this.emitEvent('criterionResultUpdated', this.formatData({}));
+    getResult(data) {
+        const validatedData = this.isValid() ? data : {};
```

Comment:
why does validatedData need to be an empty object here? Could it be `undefined`?

Reply:
I'm not entirely sure. This is a direct-ish copy and I didn't want to mess anything up. I think we can request and get results without "good" data and it should react accordingly. There is a slight chance this does something more correct for the cases where conditions are fully formed though.

I just left this relatively unchanged but moved it around for clarity.